index.js

```diff
@@ -23,13 +28,56 @@ module.exports = class ReadyResource extends EventEmitter {
     return this.closing
   }
 
+  async suspend () {
+    this.shouldBeSuspended = true
+    if (!this.opened) await this.ready()
```

Comment:
Perhaps we should instantly return if we're closed already too? (likewise for suspend)

Reply:
it does this now